src/hackney_connect.erl

```diff
@@ -66,7 +66,7 @@ create_connection(Transport, Host, Port, Options, Dynamic)
   MaxBody = proplists:get_value(max_body, Options),
 
   %% get mod metrics
-  {ok, Engine} = application:get_env(hackney, metrics),
+  Engine = application:get_env(hackney, metrics, nil),
```

Comment:
that doesn't sound good, it should be already initialized, if not raises an error